test/middleware-test.js

```diff
@@ -1933,4 +1933,66 @@ describe('input-validation middleware tests', function () {
                 });
         });
     });
+    describe('FormData', function () {
+        var app;
+        before(function () {
+            return require('./test-server-formdata').then(function (testServer) {
+                app = testServer;
+            });
+        });
+        it('only required files exists should pass', function (done) {
+            request(app)
+                .post('/pets/import')
+                .set('api-version', '1.0')
+                .attach('sourceFile', 'LICENSE')
+                .expect(200, function (err, res) {
+                    if (err) {
+                        throw err;
+                    }
+                    expect(res.body.result).to.equal('OK');
+                    done();
+                });
+        });
+        it('required and optional files exists should pass', function (done) {
+            request(app)
+                .post('/pets/import')
+                .set('api-version', '1.0')
+                .attach('sourceFile', 'LICENSE')
+                .attach('optionalFile', 'LICENSE')
+                .expect(200, function (err, res) {
+                    if (err) {
+                        throw err;
+                    }
+                    expect(res.body.result).to.equal('OK');
+                    done();
+                });
+        });
+        it('missing required file should fail', function (done) {
+            request(app)
+                .post('/pets/import')
+                .set('api-version', '1.0')
+                .attach('sourceFile1', 'LICENSE')
+                .expect(400, function (err, res) {
+                    if (err) {
+                        throw err;
+                    }
+                    expect(res.body.more_info).to.includes('body/files Missing required files: sourceFile');
+                    done();
+                });
+        });
+        it('extra files exists but not allowed should fail', function (done) {
+            request(app)
+                .post('/pets/import')
+                .set('api-version', '1.0')
+                .attach('sourceFile1', 'LICENSE')
+                .attach('sourceFile', 'LICENSE')
+                .expect(400, function (err, res) {
+                    if (err) {
+                        throw err;
+                    }
+                    expect(res.body.more_info).to.includes('body/files Extra files are not allowed. allowed files: sourceFile,optionalFile');
+                    done();
+                });
+        });
+    });
 });
```

Comment:
missing newline

Reply:
we can argue about styling :) but all the file has the same newline format, so i will keep it like that if I may

Reply:
Just a habit from work, sorry :)